codecs/jxl/enc/jxl_enc.cpp

```diff
@@ -0,0 +1,57 @@
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#include "jxl/enc_file.h"
+
+using namespace emscripten;
+
+thread_local const val Uint8Array = val::global("Uint8Array");
+
+struct JXLOptions {
+  // 1 = slowest
+  // 7 = fastest
+  int speed;
+};
+
+val encode(std::string image, int width, int height, JXLOptions options) {
+  jxl::CompressParams cparams;
+  jxl::PassesEncoderState passes_enc_state;
+  jxl::CodecInOut io;
+  jxl::PaddedBytes bytes;
+  jxl::ImageBundle* main = &io.Main();
+
+  cparams.speed_tier = static_cast<jxl::SpeedTier>(options.speed);
+  cparams.color_transform = jxl::ColorTransform::kNone;
+
```

Comment:
You'll also want an option to set the target quality (`cparams.butteraugli_distance`) – it has a useful range from say 0.5 (near-lossless) to 8 (very lossy); the default is 1 which is one unit of just-noticeable-difference.

Reply:
Added!